File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int index = nums.length-1;
        for(int i=nums.length-2;i>=0;i--){
            if(index <= nums[i] + i)
                index = i;
        }
        return index == 0;
    }
}

File: 0055-jump-game/0055-jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let indexTarget = nums.length-1;
    for(let i=nums.length-2;i>=0;i--){
        if(indexTarget <= nums[i] + i){
            indexTarget = i;
        }
    }
    return indexTarget == 0;
};